t/Event/CachedBuildFinished.t

```diff
@@ -0,0 +1,114 @@
+use strict;
+use warnings;
+use Setup;
+
+my %ctx = test_init();
+
+require Hydra::Schema;
+require Hydra::Model::DB;
+use Hydra::Event;
+use Hydra::Event::CachedBuildFinished;
+
+use Test2::V0;
+use Test2::Tools::Exception;
+use Test2::Tools::Mock qw(mock_obj);
+
+my $db = Hydra::Model::DB->new;
+hydra_setup($db);
+
+subtest "Parsing" => sub {
+    like(
+        dies { Hydra::Event::parse_payload("cached_build_finished", "") },
+        qr/takes two arguments/,
+        "empty payload"
+    );
+    like(
+        dies { Hydra::Event::parse_payload("cached_build_finished", "abc123") },
+        qr/takes two arguments/,
+        "missing the build ID"
+    );
+
+    like(
+        dies { Hydra::Event::parse_payload("cached_build_finished", "123\t456\t789\t012\t345") },
+        qr/takes two arguments/,
+        "too many arguments"
+    );
+    like(
+        dies { Hydra::Event::parse_payload("cached_build_finished", "abc123\tdef456") },
+        qr/should be integers/,
+        "evaluation ID should be an integer"
+    );
+    like(
+        dies { Hydra::Event::parse_payload("cached_build_finished", "123\tabc123") },
+        qr/should be integers/,
+        "build ID should be an integer"
+    );
+    is(
+        Hydra::Event::parse_payload("cached_build_finished", "123\t456"),
+        Hydra::Event::CachedBuildFinished->new(123, 456),
+        "one dependent build"
+    );
+};
+
+my $project = $db->resultset('Projects')->create({name => "tests", displayname => "", owner => "root"});
+my $jobset = createBaseJobset("basic", "basic.nix", $ctx{jobsdir});
+ok(evalSucceeds($jobset),               "Evaluating jobs/basic.nix should exit with return code 0");
+is(nrQueuedBuildsForJobset($jobset), 3, "Evaluating jobs/basic.nix should result in 3 builds");
+
+subtest "interested" => sub {
+    my $event = Hydra::Event::CachedBuildFinished->new(123, 456);
+
+    subtest "A plugin which does not implement the API" => sub {
+        my $plugin = {};
+        my $mock = mock_obj $plugin => ();
+
+        is($event->interestedIn($plugin), 0, "The plugin is not interesting.");
+    };
+
+    subtest "A plugin which does implement the API" => sub {
+        my $plugin = {};
+        my $mock = mock_obj $plugin => (
+            add => [
+                "cachedBuildFinished" => sub {}
+            ]
+        );
+
+        is($event->interestedIn($plugin), 1, "The plugin is interesting.");
+    };
+};
+
+subtest "load" => sub {
+    my ($build) = $db->resultset('Builds')->search(
+      { },
+      { limit => 1 }
+    )->single;
```

Comment:
Really minor nit, but is there any reason this needs to be split across lines like so?

Reply:
:+1: Nope, probably was trying different queries and got here. Fixed!